pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -175,6 +179,39 @@ type AMISelectorTerm struct {
 	Owner string `json:"owner,omitempty"`
 }
 
+// CapacityReservationSelectorTerm defines selection logic for a Capacity Reservation used by Karpenter to launch nodes.
+// If multiple fields are used for selection, the requirements are ANDed.
+type CapacityReservationSelectorTerm struct {
```

Comment:
How many of these selector fields do we think we truly need for the initial run? We can always add more fields later but the second we introduce something, we're stuck with it. High-level, I can see tags, id, and instance type making sense. Do we think the other fields would be commonly used?